src/kikuchipy/signals/util/_master_pattern.py

```diff
@@ -198,34 +176,31 @@ def _get_direction_cosines_for_fixed_pc(
     This function is optimized with Numba, so care must be taken with
     array shapes and data types.
     """
-    nrows_array = np.arange(nrows - 1, -1, -1)
-    ncols_array = np.arange(ncols)
+    x_scale = (gnomonic_bounds[1] - gnomonic_bounds[0]) / ncols
+    y_scale = (gnomonic_bounds[3] - gnomonic_bounds[2]) / nrows
 
-    # Bruker to EMsoft's v5 PC convention
-    xpc = ncols * (0.5 - pcx)
-    ypc = nrows * (0.5 - pcy)
-    zpc = nrows * pcz
+    det_gn_x = np.arange(gnomonic_bounds[0], gnomonic_bounds[1], x_scale)
 
-    det_x = xpc + (1 - ncols) * 0.5 + ncols_array
-    det_y = ypc - (1 - nrows) * 0.5 - nrows_array
-
-    ca, sa, cw, sw = _get_cosine_sine_of_alpha_and_azimuthal(
-        sample_tilt=sample_tilt,
-        tilt=tilt,
-        azimuthal=azimuthal,
-    )
-    Ls = -sw * det_x + zpc * cw
-    Lc = cw * det_x + zpc * sw
+    det_gn_y = np.arange(gnomonic_bounds[3], gnomonic_bounds[2], -y_scale)
 
     idx_1d = np.arange(nrows * ncols)[signal_mask]
     rows = idx_1d // ncols
     cols = np.mod(idx_1d, ncols)
+
     n_pixels = idx_1d.size
     r_g_array = np.zeros((n_pixels, 3), dtype=np.float64)
+
+    x_half_step = np.true_divide(x_scale, 2)
+    y_half_step = np.true_divide(y_scale, 2)
+
     for i in nb.prange(n_pixels):
-        r_g_array[i, 0] = det_y[rows[i]] * ca + sa * Ls[cols[i]]
-        r_g_array[i, 1] = Lc[cols[i]]
-        r_g_array[i, 2] = -sa * det_y[rows[i]] + ca * Ls[cols[i]]
+        r_g_array[i, 0] = (det_gn_x[cols[i]] + x_half_step) * pcz
+        r_g_array[i, 1] = (det_gn_y[rows[i]] - y_half_step) * pcz
+        r_g_array[i, 2] = pcz
+
+    # rotate vectors from detector reference frame CSd to
```

Comment:
I'd remove the comment and rather rename the orientation matrix variable to be self-explanatory.